stl/inc/xstring

```diff
@@ -3153,45 +3241,55 @@ public:
+    _CONSTEXPR20_CONTAINER basic_string& insert(const size_type _Off, const _StringViewIsh& _Right) {
+        // insert _Right at _Off
         const basic_string_view<_Elem, _Traits> _As_view = _Right;
         return insert(_Off, _As_view.data(), _Convert_size<size_type>(_As_view.size()));
     }
 
     template <class _StringViewIsh, _Is_string_view_ish<_StringViewIsh> = 0>
-    basic_string& insert(const size_type _Off, const _StringViewIsh& _Right, const size_type _Roff,
-        const size_type _Count = npos) { // insert _Right [_Roff, _Roff + _Count) at _Off
+    _CONSTEXPR20_CONTAINER basic_string& insert(
+        const size_type _Off, const _StringViewIsh& _Right, const size_type _Roff, const size_type _Count = npos) {
+        // insert _Right [_Roff, _Roff + _Count) at _Off
         basic_string_view<_Elem, _Traits> _As_view = _Right;
         return insert(_Off, _As_view.substr(_Roff, _Count));
     }
 #endif // _HAS_CXX17
 
-    basic_string& insert(
+    _CONSTEXPR20_CONTAINER basic_string& insert(
         const size_type _Off, _In_reads_(_Count) const _Elem* const _Ptr, _CRT_GUARDOVERFLOW const size_type _Count) {
         // insert [_Ptr, _Ptr + _Count) at _Off
         _Mypair._Myval2._Check_offset(_Off);
         const size_type _Old_size = _Mypair._Myval2._Mysize;
-        if (_Count <= _Mypair._Myval2._Myres - _Old_size) {
-            _Mypair._Myval2._Mysize = _Old_size + _Count;
-            _Elem* const _Old_ptr   = _Mypair._Myval2._Myptr();
-            _Elem* const _Insert_at = _Old_ptr + _Off;
-            // the range [_Ptr, _Ptr + _Ptr_shifted_after) is left alone by moving the suffix out,
-            // while the range [_Ptr + _Ptr_shifted_after, _Ptr + _Count) shifts down by _Count
-            size_type _Ptr_shifted_after;
-            if (_Ptr + _Count <= _Insert_at
-                || _Ptr > _Old_ptr + _Old_size) { // inserted content is before the shifted region, or does not alias
-                _Ptr_shifted_after = _Count; // none of _Ptr's data shifts
-            } else if (_Insert_at <= _Ptr) { // all of [_Ptr, _Ptr + _Count) shifts
-                _Ptr_shifted_after = 0;
-            } else { // [_Ptr, _Ptr + _Count) contains _Insert_at, so only the part after _Insert_at shifts
-                _Ptr_shifted_after = static_cast<size_type>(_Insert_at - _Ptr);
-            }
 
-            _Traits::move(_Insert_at + _Count, _Insert_at, _Old_size - _Off + 1); // move suffix + null down
-            _Traits::copy(_Insert_at, _Ptr, _Ptr_shifted_after);
-            _Traits::copy(
-                _Insert_at + _Ptr_shifted_after, _Ptr + _Count + _Ptr_shifted_after, _Count - _Ptr_shifted_after);
-            return *this;
+        // checking for overlapping ranges is technically UB (considering string literals), so just always reallocate
+        // and copy to the new buffer if constant evaluated
+#ifdef __cpp_lib_constexpr_string

```

Comment:
I think it might be clearer if we would again use a local boolean like above and avoid the additional indentation.

Or better said we should focus on one technique